data/crac-creation/crac-creator-cse/src/main/java/com/farao_community/farao/data/crac_creation/creator/cse/remedial_action/TRemedialActionAdder.java

```diff
@@ -198,81 +195,85 @@ private void importHvdcRangeAction(TRemedialAction tRemedialAction) {
+                            ImportStatus.INCONSISTENCY_IN_DATA,
+                            String.format("HVDC %s has %s (>1) HVDC nodes", raId, tRemedialAction.getHVDCRange().getHVDCNode().size()),
+                            hvdcNodes.getFromNode().getV(),
+                            hvdcNodes.getToNode().getV()));
             return;
         }
-        // Temporary bypass : get network Identifiable if HVDC is not found in network with UcteHvdcElementHelper
-        String hvdcId;
-        boolean isInverted = false;
-        if (hvdcHelper.isValid()) {
-            hvdcId = hvdcHelper.getIdInNetwork();
-            isInverted = hvdcHelper.isInvertedInNetwork();
-        } else {
-            Identifiable<?> hvdcIdentifiable = network.getIdentifiable(String.format("%s %s %s", hvdcNode.getFromNode().getV(), hvdcNode.getToNode().getV(), String.valueOf(hvdcNode.getOrder().getV())));
-            if (hvdcIdentifiable == null) {
-                cseCracCreationContext.addRemedialActionCreationContext(CseHvdcCreationContext.notImported(tRemedialAction, hvdcHelper.getUcteId(), ImportStatus.ELEMENT_NOT_FOUND_IN_NETWORK, hvdcHelper.getInvalidReason()));
-                return;
-            }
-            if (hvdcIdentifiable instanceof HvdcLine) {
-                hvdcId = hvdcIdentifiable.getId();
-            } else {
-                hvdcIdentifiable = network.getIdentifiable(String.format("%s %s %s", hvdcNode.getToNode().getV(), hvdcNode.getFromNode().getV(), String.valueOf(hvdcNode.getOrder().getV())));
-                if (hvdcIdentifiable instanceof HvdcLine) {
-                    hvdcId = hvdcIdentifiable.getId();
-                    isInverted = true;
-                } else {
-                    cseCracCreationContext.addRemedialActionCreationContext(CseHvdcCreationContext.notImported(tRemedialAction, hvdcHelper.getUcteId(), ImportStatus.OTHER, hvdcHelper.getInvalidReason()));
-                    return;
-                }
-            }
-        }
 
-        // Invert ranges if isInverted
-        double minRange = isInverted ? -tRemedialAction.getHVDCRange().getMax().getV() : tRemedialAction.getHVDCRange().getMin().getV();
-        double maxRange = isInverted ? -tRemedialAction.getHVDCRange().getMin().getV() : tRemedialAction.getHVDCRange().getMax().getV();
+        // ---- check if generator are presents
+        if (!generatorFromHelper.isValid() || !generatorToHelper.isValid()) {
+            GeneratorHelper invalidGenerator = generatorFromHelper.isValid() ? generatorToHelper : generatorFromHelper;
+            cseCracCreationContext.addRemedialActionCreationContext(
+                    CseHvdcCreationContext.notImported(tRemedialAction,
+                            ImportStatus.ELEMENT_NOT_FOUND_IN_NETWORK,
+                            invalidGenerator.getDetail(),
+                            hvdcNodes.getFromNode().getV(),
+                            hvdcNodes.getToNode().getV()));
+            return;
```

Comment:
if both are invalid, do we want to add both contexts ?

Reply:
we add one context with the two values of the CRAC (fromNode, toNode), but with any of the associated generators, as they might have not been found in the network.